### ISSUE TITLE
Help with Authentik config for External Redis server with TLS

### ISSUE THREAD

**Issue by tltglitch** [goauthentik/authentik#10092]
I am using the latest version of authentik and i am trying to use a external redis server. Can someone help me with the correct docker compose file to make this work ? 

Below is my config and no matter what i do it doesnt work. The images come up unhealthy.

---
services:
  postgresql:
    image: docker.io/library/postgres:16-alpine
    container_name: tlt-authentik-sso-project-postgresql
    restart: unless-stopped
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -d $${POSTGRES_DB} -U $${POSTGRES_USER}"]
      start_period: 20s
      interval: 30s
      retries: 5
      timeout: 5s
    volumes:
      - database:/var/lib/postgresql/data
    environment:
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_DB: ${POSTGRES_DB}
    env_file:
      - .env

  server:
    image: ${AUTHENTIK_IMAGE:-ghcr.io/goauthentik/server}:${AUTHENTIK_TAG:-2024.4.2}
    container_name: tlt-authentik-sso-project-server
    restart: unless-stopped
    command: server
    environment:
      AUTHENTIK_REDIS__HOST: redis
      AUTHENTIK_REDIS__PORT: 6379
      AUTHENTIK_REDIS__PASSWORD: ${REDIS_PASSWORD}
      AUTHENTIK_REDIS__TLS:  "true"
      AUTHENTIK_REDIS__TLS_REQS: "true"
      AUTHENTIK_REDIS__TLS_CA_CERT: /etc/ssl/redis/redis-server-cert.pem
      AUTHENTIK_POSTGRESQL__HOST: postgresql
      AUTHENTIK_POSTGRESQL__USER: ${POSTGRES_USER}
      AUTHENTIK_POSTGRESQL__NAME: ${POSTGRES_DB}
      AUTHENTIK_POSTGRESQL__PASSWORD: ${POSTGRES_PASSWORD}
    volumes:
      - ./media:/media
      - ./custom-templates:/templates
    env_file:
      - .env
    ports:
      - "${COMPOSE_PORT_HTTP:-9000}:9000"
      - "${COMPOSE_PORT_HTTPS:-9443}:9443"
    depends_on:
      - postgresql

  worker:
    image: ${AUTHENTIK_IMAGE:-ghcr.io/goauthentik/server}:${AUTHENTIK_TAG:-2024.4.2}
    restart: unless-stopped
    command: worker
    environment:
      AUTHENTIK_REDIS__HOST: redis
      AUTHENTIK_REDIS__PORT: 6379
      AUTHENTIK_REDIS__PASSWORD: ${REDIS_PASSWORD}
      AUTHENTIK_REDIS__TLS:  "true"
      AUTHENTIK_REDIS__TLS_REQS: "true"
      AUTHENTIK_REDIS__TLS_CA_CERT: /etc/ssl/redis/redis-server-cert.pem
      AUTHENTIK_POSTGRESQL__HOST: postgresql
      AUTHENTIK_POSTGRESQL__USER: ${POSTGRES_USER}
      AUTHENTIK_POSTGRESQL__NAME: ${POSTGRES_DB}
      AUTHENTIK_POSTGRESQL__PASSWORD: ${POSTGRES_PASSWORD}
    user: root
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./media:/media
      - ./certs:/certs
      - ./custom-templates:/templates
    env_file:
      - .env
    depends_on:
      - postgresql

volumes:
  database:
    driver: local

What am i doing wrong?

**Comment by dhruv-bardolia** [comment]
@tltglitch , I am also encountering the same issue. I am deploying it in a Kubernetes cluster. Have you been able to resolve this problem? If so, could you please share the necessary changes?

**Comment by tltglitch** [comment]
@dhruv-bardolia yes i did and i hope this helps you as well. I amusing .env file for tokens, so below is my redis config for the server and worker path for authentik:

 AUTHENTIK_REDIS__HOST: 1.1.1.1
      AUTHENTIK_REDIS__PORT: 6379
      AUTHENTIK_REDIS__PASSWORD: ${REDIS_PASSWORD}
      AUTHENTIK_REDIS__TLS: true

I am not using a user for redis just a password, you can add a user in the config if your environment uses that. This works for me. Please let me know if you have anymore questions

**Comment by mike-pisman** [comment]
@tltglitch, what did you use for `redis-server-cert.pem`? I am having a similar issue #12041

**Comment by ordidaad** [comment]
Does it supports prefix and database number?

**Comment by rissson** [comment]
Closing since we don't use redis anymore since 2025.10
